src/leaspy/io/data/covariate_dataframe_data_reader.py

```diff
@@ -0,0 +1,245 @@
+
+__all__ = ["CovariateDataframeDataReader"]
+
+
+class CovariateDataframeDataReader(AbstractDataframeDataReader):
+    """
+    Methods to convert :class:`pandas.DataFrame` to `Leaspy`-compliant data containers for longitudinal data with covariates.
+
+    Parameters
+    ----------
+    covariate_names: List[str]
+        Names of the columns in dataframe that contains the covariates
+
+    Raises
+    ------
+    :exc:`.LeaspyDataInputError`
+    """
+
+    def __init__(
+        self,
+        *,
+        covariate_names: list[str],
+    ):
+        super().__init__()
+        if not covariate_names:
+            raise LeaspyDataInputError("You must prrovide at least one covariate name.")
+        self.covariate_names = covariate_names
+        self.visit_reader = VisitDataframeDataReader()
+
+    @property
+    def long_outcome_names(self) -> list[FeatureType]:
+        """Name of the longitudinal outcomes in dataset"""
+        return self.visit_reader.long_outcome_names
+
+    @property
+    def n_visits(self) -> int:
+        """Number of visit in the dataset"""
+        return self.visit_reader.n_visits
+
+    ######################################################
+    #               COVARIATE METHODS
+    ######################################################
+
+    def _check_headers(self, columns: list[str]) -> None:
+        """
+        Check mendatory dataframe headers
+
+        Parameters
+        ----------
+        columns: List[str]
+            Names of the columns headers of the dataframe that contains patients information
+        """
+        self.visit_reader._check_headers(columns)
+
+    def _set_index(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Set the index suited for the type of information contained in the dataframe
+
+        Parameters
+        ----------
+        df: pd.DataFrame
+            Dataframe with patient information
+
+        Returns
+        -------
+        df: pd.DataFrame
+            Dataframe with the right index
+        """
+
+        return self.visit_reader._set_index(df)
+
+    def _clean_dataframe_covariates(
+        self, df: pd.DataFrame, *, drop_full_nan: bool, warn_empty_column: bool
+    ) -> pd.DataFrame:
+        """
+        Clean the dataframe that contains patient information
+
+        Parameters
+        ----------
+        df: pd.DataFrame
+            Dataframe with patient information
+
+        drop_full_nan: bool
+            If set to True, raw full of nan are dropped
+
+        warn_empty_column: bool
+            If set to True, a warning is raise for columns full of nan
+
+
+        Returns
+        -------
+        df: pd.DataFrame
+            Dataframe with clean information
+        """
+
+        # [SPECIFIC] check_available_data
+        df_covariate = df.copy(deep=True)
+
+        # Assert covariates columns are the only one available
+        assert (df_covariate.columns == self.covariate_names).all()
```

Comment:
Usually, we use assertions in tests and prefer proper error raising in code because it gives much more details to the end user. So I'd suggest something like that:

```suggestion
        if not (df_covariate.columns == self.covariate_names).all():
            raise LeaspyDataInputError(
                f"The covariate column names {df_covariate.columns} are "
                f"different from the provided covariate names {self.covariate_names}."
            )
```